sound/soc/sof/intel/hda-stream.c

```diff
@@ -570,14 +587,18 @@ static bool hda_dsp_stream_check(struct hdac_bus *bus, u32 status)
 			snd_hdac_stream_writeb(s, SD_STS, sd_status);
 
 			active = true;
-			if (!s->substream ||
+			if (!(s->substream || s->cstream) ||
 			    !s->running ||
 			    (sd_status & SOF_HDA_CL_DMA_SD_INT_COMPLETE) == 0)
 				continue;
 
 			/* Inform ALSA only in case not do that with IPC */
-			if (sof_hda->no_ipc_position)
+			if (s->substream && sof_hda->no_ipc_position) {
```

Comment:
that check looks odd to me. is it needed? We cannot have the compress stream use the DMA information, can we?

Reply:
Updated commit:
ASoC: SOF: Intel: Account for compress streams when servicing IRQs
as indeed it really does not look like compress-specific, although we might also want to revert this later on - s->substream check was there to not invoke _set_bytes_transferred unnecessarily for PCM streams as position assignment performed by it is completely ignored by said stream handlers.